### PR TITLE
Add ActionCacheStatistics to BEP

### PR DESCRIPTION
**Pull request by crydell-ericsson**
Solves #17315 .

The field action_cache_statistics has been added to the ActionSummary message of the BuildMetrics message in the build event protocol. This field is defined with the already-existing ActionCacheStatistics message and is set in the MetricsCollector when the action cache is saved.

### CODE REVIEW

**Comment by brentleyjones** [comment]
@bazelbuild/triage Can we get someone to review this please?

**Comment by crydell-ericsson** [comment]
Are there any updates on the review of this PR?

**Comment by brentleyjones** [comment]
@bazelbuild/triage @michaeledgar Can we get a review, or assigned to someone else, please? Thanks.

**Review by haxorz — APPROVED** [review]

**Comment by brentleyjones** [comment]
@bazel-io flag

**Comment by iancha1992** [comment]
@bazel-io fork 6.3.0

**Comment by iancha1992** [comment]
@brentleyjones @haxorz @crydell-ericsson 
I'm trying to cherry-pick this to release-6.3.0, but there are some conflicts. 

1) src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD
"//src/main/java/com/google/devtools/build/lib/packages/metrics:package_load_metrics_proto" is missing as a dependency for "build_event_stream_proto". May need another commit that adds this to the file.

2) src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto
"import "src/main/java/com/google/devtools/build/lib/packages/metrics/package_load_metrics.proto" is missing. May need another commit that adds this import. 

3) src/main/java/com/google/devtools/build/lib/metrics/BUILD
For "metrics_module", it is missing "//third_party:auto_value" as one of its dependencies. May need another commit that adds this.

cc: @bazelbuild/triage

**Comment by brentleyjones** [comment]
I'll leave it to @crydell-ericsson on how easy it is to get the dependent commits in, or if this can only really work with 7.0. The change looked small enough, so I was hopeful.

**Comment by crydell-ericsson** [comment]
All of the differences mentioned by @iancha1992 are changes that have ended up being added while keeping this PR up to date with the master branch (perhaps that's something I shouldn't have done?). If you look at the initial commit in this PR [5f68c8f](https://github.com/bazelbuild/bazel/pull/17615/commits/5f68c8f9479bd76ef64cad80e5ea156aa97397e9), you'll see that it does not have any of these dependencies.

In other words, the listed dependencies are not needed for the commit, they just happened to cause conflicts by being right next to the dependencies that were needed. Should I do anything from my side to amend this?

**Comment by brentleyjones** [comment]
I think you can submit the cherry-pick PR (based on your original commit maybe?) and @iancha1992 would accept that.

**Comment by crydell-ericsson** [comment]
I have put up a cherry pick PR in #18914 now. I haven't submitted PRs for cherry picks before, let me know if I've done anything incorrectly and I'll try to fix it.

**Comment by haxorz** [comment]
I approved. The diff looked fine.

I've never authored or reviewed a cherrypick before, so someone else should probably double-check.
